### PR TITLE
Implement Monthly-checking Feature for Mentees, Mentors and Admins

### LINKED ISSUE #198: Introduce Monthly Check-In Feature for Mentees, Mentors, and Admins
**Is your feature request related to a problem? Please describe.**

 I'm always frustrated when the process of submitting and reviewing monthly check-ins for mentees is inefficient and lacks a streamlined workflow. Currently, there is no integrated system for mentees to submit their progress, for mentors to provide feedback, and for admins to monitor the overall progress.

**Describe the solution you'd like**

 I would like to introduce a new feature that includes:

- Mentee Submission Form: A form for mentees to submit their monthly progress, including general updates, progress towards goals, and media content links.
- Mentor View: A view for mentors to see the submitted check-ins and provide feedback, including marking the check-in as reviewed.
- Admin View: A view for admins to oversee the check-in process, ensuring that mentors are providing timely feedback and that mentees are completing their check-ins.

**Describe alternatives you've considered**

- Using external tools for feedback submission, which is not integrated with the current system.
- Manual tracking of feedback and progress, which is error-prone and inefficient.
- Periodic email updates, which lack real-time interaction and tracking.
**Additional context**
Add any other context or screenshots about the feature request here

### PR DESCRIPTION
**Pull request by mayura-andrew**
## Purpose

The purpose of this PR is to fix #198  by introducing a comprehensive monthly check-in feature for mentees, mentors, and admins.

## Goals

- Mentees: Allow mentees to submit their monthly progress, view mentor feedback, and see the history of their submissions.
- Mentors: Enable mentors to view their mentees' submissions, provide feedback, and track the history of check-ins.
- Admins: Provide admins with the ability to oversee the progress of all mentees, ensuring that mentors are providing timely feedback and that mentees are completing their check-ins.

## Approach

- Implemented a submission form UI for mentees to submit their monthly progress.
- Created a view for mentors to see the submitted check-ins and provide feedback.
- Developed an admin dashboard to monitor the progress of all mentees.
- Integrated the feedback form into the mentor dashboard.
- Integrated the check-in status view into the mentee dashboard.
- Integrated the admin dashboard view to oversee the check-in process.
- Added necessary hooks and schemas for form validation and submission.

## Screenshots

New Mentor Dashboard 

image

New Mentee Dashboard UI 

image

image

image

image

image

image

image

Mentee Monthly-checking Form Modal

image

Mentor Feedback UI 

image

image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Test environment
 
NodeJS: 20.15.0
FireFox: 129.0
## Learning

- Researched best practices for form validation using react-hook-form and zod.
- Reviewed UI/UX patterns for feedback forms and admin dashboards.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,16 +19,20 @@
...
+    "date-fns": "^4.1.0",
     "dotenv": "^16.0.3",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-hook-form": "^7.51.0",
     "react-intersection-observer": "^9.13.1",
     "react-router": "^6.11.2",
     "react-router-dom": "^6.11.2",
+    "react-spring": "^9.7.4",
+    "react-tag-input": "^6.10.3",
```
Comment (anjula-sack) [inline package.json]: We are not using this right?

File: src/components/TagInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+interface TagInputProps {
+  tags: string[];
+  setTags: React.Dispatch<React.SetStateAction<string[]>>;
+  maxTags: number;
+  register: UseFormRegister<any>;
+  onValidationError?: (error: string) => void;
+}
+
+const TagInput: React.FC<TagInputProps> = ({
```
Comment (anjula-sack) [inline src/components/TagInput/index.tsx]: Remove this unused component

File: package.json [hunk 3]
```diff
@@ -19,16 +19,20 @@
...
     "axios": "^1.4.0",
+    "date-fns": "^4.1.0",
     "dotenv": "^16.0.3",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-hook-form": "^7.51.0",
     "react-intersection-observer": "^9.13.1",
     "react-router": "^6.11.2",
     "react-router-dom": "^6.11.2",
+    "react-spring": "^9.7.4",
```
Comment (anjula-sack) [inline package.json]: Remov this library as well. We don't need animations for now.

File: src/components/CongratsModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}) => {
+  const animation = useSpring({
+    opacity: isOpen ? 1 : 0,
+    transform: isOpen ? 'translateY(0)' : 'translateY(-50%)',
+  });
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="fixed inset-0 flex items-center justify-center z-50 bg-black bg-opacity-50">
```
Comment (anjula-sack) [inline src/components/CongratsModal/index.tsx]: Try to reuse the existing modal component for this

File: src/components/MonthlyChecking/MenteeCheckInFormModal.component.tsx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+                <option value="March">March</option>
+                <option value="April">April</option>
+                <option value="May">May</option>
+                <option value="June">June</option>
+                <option value="July">July</option>
+                <option value="August">August</option>
+                <option value="September">September</option>
+                <option value="October">October</option>
+                <option value="November">November</option>
+                <option value="December">December</option>
```
Comment (anjula-sack) [inline src/components/MonthlyChecking/MenteeCheckInFormModal.component.tsx]: Create a loop for this

const months = ['January', 'February', 'March' ....]

months.map()

File: src/components/MonthlyChecking/MonthlyCheckingHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+            <>
+              <ArrowDownIcon />
+              Guidelines
+            </>
+          ) : (
+            <>
+              <ArrowRightIcon />
+              Guidelines
+            </>
+          )}
```
Comment (anjula-sack) [inline src/components/MonthlyChecking/MonthlyCheckingHeader.tsx]:
```suggestion
{showGuidelines ? <ArrowDownIcon /> : <ArrowRightIcon />} Guidelines
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/CongratsModal/index.tsx (hunk 4 above)
Comment (mayura-andrew) [reply src/components/CongratsModal/index.tsx]: this is not using, i'll remove this

File: src/App.tsx [hunk 7]
```diff
@@ -39,16 +37,12 @@ const App: React.FC = () => {
...
-                  element={<MenteeApplications />}
-                />
+                <Route path="/mentor/dashboard/*" element={<MyMentees />} />
                 <Route path="/mentee/dashboard" element={<MenteeDashboard />} />
-                <Route path="/mentor/my-mentees/*" element={<MyMentees />} />
                 <Route path="/mentors" element={<Mentors />} />
                 <Route path="/mentors/:mentorId" element={<MentorProfile />} />
                 <Route path="/mentees/:menteeId" element={<MenteeProfile />} />
-                <Route path="/mentee-checkin" element={<MenteeCheckIn />} />
+
```
Comment (anjula-sack) [inline src/App.tsx]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: package.json [hunk 8]
```diff
@@ -19,9 +19,11 @@
   "license": "ISC",
   "dependencies": {
     "@hookform/resolvers": "^3.3.4",
+    "@react-spring/web": "^9.7.4",
```
Comment (anjula-sack) [inline package.json]: Remove this
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/MentorProfile/MentorProfile.component.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import ProfilePic from '../ProfilePic/index.tsx';
+import { type Mentor } from '../../types.ts';
+
+interface MentorCardProps {
+  mentor: Mentor;
+}
+
+const MentorProfileCard: React.FC<MentorCardProps> = ({ mentor }) => {
```
Comment (anjula-sack) [inline src/components/MentorProfile/MentorProfile.component.tsx]: Can't we reuse the existing mentor card?

File: src/components/NotificationBadge/index.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+              fontWeight="bold"
+              fill={textColor}
+              textAnchor="middle"
+              dominantBaseline="central"
+            >
+              {displayCount}
+            </text>
+          </>
+        )}
+      </svg>
```
Comment (anjula-sack) [inline src/components/NotificationBadge/index.tsx]: Move this to svgs

File: src/components/Spinner/Spinner.component.tsx [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
+
+const Spinner: React.FC = () => (
```
Comment (anjula-sack) [inline src/components/Spinner/Spinner.component.tsx]: Use this https://github.com/sef-global/scholarx-frontend/blob/development/src/assets/svg/Loading.tsx

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/assets/svg/Icons/NotificationBellSVG.tsx [hunk 12]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { NotificationProps } from '../../../types';
+
+const NotificationBellSVG: React.FC<NotificationProps> = ({
```
Comment (anjula-sack) [inline src/assets/svg/Icons/NotificationBellSVG.tsx]:
```suggestion
const NotificationBell: React.FC<NotificationProps> = ({
```
Rename the component folder as well `NotificationBell`

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Good job! @mayura-andrew

**Pull request merged**